qa/tasks/ceph_iscsi.py

```diff
@@ -0,0 +1,159 @@
+"""
+Run ceph-iscsi cluster setup
+"""
+import logging
+import socket
+import re
+import contextlib
+from io import StringIO
+from teuthology.exceptions import CommandFailedError, ConnectionLostError
+from teuthology.orchestra import run
+from textwrap import dedent
+
+log = logging.getLogger(__name__)
+
+class IscsiSetup(object):
+    def __init__(self, ctx, config):
+        self.ctx = ctx
+        self.config = config
+        self.target_iqn = "iqn.2003-01.com.redhat.iscsi-gw:ceph-gw"
+        self.client_iqn = "iqn.1994-05.com.redhat:client"
+        self.trusted_ip_list = []
+        self.background_procs = []
+
+    # Copied from ceph-iscsi project
+    def get_host(self, ip=''):
+        """
+        If the 'ip' is empty, it will return local node's
+        hostname, or will return the specified node's hostname
+        """
+        return socket.getfqdn(ip)
+
+    # Get the trusted ip list from the ceph.conf
+    def _get_trusted_ip_list(self, remote, target):
+        if self.trusted_ip_list is not []:
+            return
+        stdout = StringIO()
+        stderr = StringIO()
+        try:
+            args = ['cat', target]
+            remote.run(args=args, stdout=stdout, stderr=stderr)
+        except CommandFailedError:
+            if "No such file or directory" in stderr.getvalue():
+                raise
+
+        ips = re.findall(r'mon host = \[v2:([\d.]+).*v1.*\],\[v2:([\d.]+).*v1.*\],\[v2:([\d.]+).*v1.*\]', stdout.getvalue())
```

Comment:
Nit: since you are going to need to find the host IP and name, perhaps just pull this data out of `remote.ip_address` and `remote.hostname`

Reply:
Sure. will do.